aten/src/ATen/native/cpu/BinaryOpsKernel.cpp

```diff
@@ -502,6 +502,80 @@ void min_elementwise_kernel(TensorIterator& iter) {
   }
 }
 
+void maximum_kernel(TensorIterator& iter) {
+  if (isIntegralType(iter.dtype(), /*includeBool=*/ true)) {
+    AT_DISPATCH_INTEGRAL_TYPES_AND(at::ScalarType::Bool, iter.dtype(), "maximum_cpu", [&] {
+      cpu_kernel(iter, [](scalar_t a, scalar_t b) -> scalar_t {
+        return a >= b ? a : b;
+      });
+    });
+  } else if (isFloatingType(iter.dtype())) {
+    AT_DISPATCH_FLOATING_TYPES_AND2(at::ScalarType::Half, at::ScalarType::BFloat16, iter.input_dtype(), "maximum_cpu", [&]() {
+      cpu_kernel(iter, [](scalar_t a, scalar_t b) -> scalar_t {
+        if (_isnan(a)) {
+          return a;
+        }
+        if (_isnan(b)) {
+          return b;
+        }
+        return a >= b ? a : b;
+      });
+    });
+  } else {
+    AT_DISPATCH_COMPLEX_TYPES(iter.dtype(), "maximum_cpu", [&] {
```

Comment:
Comparisons between complex numbers aren't well-defined mathematically and we prohibit them elsewhere (NumPy is updating their functionality to conform to ours). I would think that means that we shouldn't support maximum and minimum for complex tensors?

Reply:
Thanks for the kind suggestions, I have added checks to prevent complex inputs and deleted redundant complex dispatch codes.